### PR TITLE
Fix chunksize not loading for peeler when launched from gui

### PR DESCRIPTION
**Pull request by jpgill86**
In `MainWindow.run_peeler`, `peeler.change_params` is called without passing in the `chunksize`. This sets the Peeler's `chunksize` to the default value of 1024.

To pass the correct `chunksize` to `peeler.change_params`, the `chunksize` must be retrieved. If the `chunksize` was set in `dialog_fullchain_params` this session, it would be possible to retrieve it from there. However, if a data set was loaded with a previously initialized catalogue, the default value found in `dialog_fullchain_params` will be incorrect.

So, to solve this problem, this commit first adds `chunksize` to `CatalogueConstructor.make_catalogue` so that it can be saved by `CatalogueConstructor.make_catalogue_for_peeler` and `DataIO.save_catalogue`. Second, the commit extracts the `chunksize` from `initial_catalogue` after it has been loaded by `DataIO.load_catalogue` inside `MainWindow.run_peeler` and passes it with the other parameters to `peeler.change_params`.

Extraction of the new `chunksize` parameter from `initial_catalogue` is wrapped in a `try` statement for backwards compatibility with old catalogues created before this fix.

### CODE REVIEW

File: tridesclous/catalogueconstructor.py [hunk 1]
```diff
@@ -1492,6 +1492,7 @@ def make_catalogue(self):
         #~ self.catalogue['cluster_colors'].update(self.colors)
         
         #params
+        self.catalogue['chunksize'] = self.info['chunksize']
```
Comment (jpgill86) [inline tridesclous/catalogueconstructor.py]: This addition ensures that the `chunksize` gets saved to file with the catalogue.

File: tridesclous/gui/mainwindow.py [hunk 2]
```diff
@@ -412,14 +412,18 @@ def run_peeler(self):
...
     1. Initialize Catalogue (if not done)
     2. Open CatalogueWindow
     3. Make catalogue for peeler
                     """.format(chan_grp)
                     errors.append(txt)
                     continue
+                try:
+                    d['chunksize'] = initial_catalogue['chunksize']
+                except KeyError as e:
+                    print('chunksize was not saved with catalogue, peeler will use default chunksize')
```
Comment (jpgill86) [inline tridesclous/gui/mainwindow.py]: This code extracts the `chunksize` from the loaded catalogue if it's available.

File: tridesclous/gui/mainwindow.py [hunk 3]
```diff
@@ -412,14 +412,18 @@ def run_peeler(self):
             #~ if True:
                 initial_catalogue = self.dataio.load_catalogue(chan_grp=chan_grp)
                 if initial_catalogue is None:
-                    txt =  """chran_grp{}
+                    txt =  """chan_grp{}
```
Comment (jpgill86) [inline tridesclous/gui/mainwindow.py]: Typo

File: tridesclous/gui/mainwindow.py [hunk 4]
```diff
@@ -431,7 +435,7 @@ def run_peeler(self):
                 
             except Exception as e:
                 print(e)
-                error = """chran_grp{}\n{}""".format(chan_grp, e)
+                error = """chan_grp{}\n{}""".format(chan_grp, e)
```
Comment (jpgill86) [inline tridesclous/gui/mainwindow.py]: Typo

**Comment by jpgill86** [comment]
Even with this fix, it's still the case that the following workflow, recommended in the example Jupyter notebook (`example_locust_dataset.ipynb`), doesn't use the correct `chunksize`:

    initial_catalogue = dataio.load_catalogue(chan_grp=0)
    peeler = Peeler(dataio)
    peeler.change_params(catalogue=initial_catalogue)

Changing the last line to

    peeler.change_params(catalogue=initial_catalogue, chunksize=initial_catalogue['chunksize'])

is equivalent to the fix proposed in this PR for the gui. However, since `initial_catalogue` is already passed in (`catalogue=initial_catalogue`), maybe `Peeler.change_params` should search the `catalogue` argument for `chunksize` instead? Maybe something like this in `Peeler.change_params`?

    if chunksize is None:  # parameter default would need to be changed from 1024 to None
        self.chunksize = catalogue['chunksize']
    else:
        self.chunksize = chunksize

Would that be a more appropriate and simpler fix which solves this problem both for the gui and for scripts?

**Comment by samuelgarcia** [comment]
Hi @jpgill86.
Thank you for having at this chunksize problem.
Thing are a bit diffivult than they appear.

I wil explain.

In actual implementation:
  * **chunksize** have no effect on the filtering steps if the **lostfront_chunksize** is big enough.
    So the catalogue process is not effected by this parameters. And this is good.
  * Unfortunatly **chunksize**  have a very small impact on the peeler process. This append because
   of the border. The fact that a spike appear at a border of a chunk is correctly done becasue a small 
   sweep is kept for the next process_schunk BUT when a spike collision append at the border it some 
   rare cases, for instance when one spike is at left and one at right, the peeler do the template matchin 
   in the wrong order and can lead to small errors.
 
So I want to have the possibility of changing **chunksize** at peeler time to:
  * test this chunksize effect on spike colision.
  * change the latency of peeler in online use cases.

So in conclusion, I am not fan of having this chunksize in catalogue because this parameters shoudl not influence the results. **lostfront_chunksize** is important.

So I prefer to explicitly to let the user choose the chunksize in the param dialog for peeler.

I propose to make a separte PR for this that include all the bug you found about this but deal the chunksize default params in another way.

OK for this ?

**Comment by samuelgarcia** [comment]
No CatalogueConstructor do not ignore chunksize but the result is not affected by this chunksize. Only the memory.

I must improve the doc about this lost_chunksize. It is one the challenge : make filtfilt processing but only forward buffer!
This idea is that the forward filter is done until the end of the new buffer. It  has no error because the internal state of filter are internally kept. It is a general case in forward filtering the chunksize do not affect the result. 
For the backward filter, for each chunk,  I start a stateless filter in reverse direction at the new rigth border and go back more than the buffer itself.  A internal fifo allow this. In that case the border is coruptted and so the available buffer is [index-lost_chunksize-chunksize:index-lost_chunksize].

It is exactly an overlap, not in the sens of overlap with window and add. It is a fifo for teh backward.

I will do the PR.

**Comment by jpgill86** [comment]
Hmm, I'm not certain that I understand, but I don't think I need to understand it completely right now. A picture showing the k-th chunk (current chunk) between the k-1 and k+1 chunks, along with their borders and an appropriate representation of the `lostfront_chunksize`, would likely be a good addition to the docs!

**Comment by samuelgarcia** [comment]
I guess this is the fault of my fuzzy english gramar.
I will do a sketch.

**Comment by samuelgarcia** [comment]
Done other way in #102.
